### PR TITLE
Fixing The Data Transfer Issue

### LINKED ISSUE #52: Data Transfering Issue
Trying to find the best way to transfer the data between the front end and back end.

### PR DESCRIPTION
**Pull request by BlindFl0W**
- Add DTO data transfer object class that contains a list of attendance records and can return a 2D array to be used in the front end.
- Edited all the related classes, interfaces and methods from useing a referance of a window object to using the DTO.

This edit  fixes #52 once and hopfully for all.

### CODE REVIEW

**Comment by BlindFl0W** [comment]
@mistaluai 
I have tested the app and the view and super view functionalty works fine and views the requested data on the front end.
Please do further testing on the DataTransferFixing branch and review the before merging.

File: src/DTO/DTO.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public List<AttendanceRecord> getAttendanceRecords() {
+        return attendanceRecords;
+    }
+
+    public void setAttendanceRecords(List<AttendanceRecord> attendanceRecords) {
+        this.attendanceRecords = attendanceRecords;
+    }
+
+    public String[][] getAttendanceRecordsArray() {
+        String[][] attendanceHistory = new String[attendanceRecords.size()][3];
```
Comment (mistaluai) [inline src/DTO/DTO.java:24]: when the size of the attendance history is equal to zero, as an example, make the array look like {{"Not Found","Not Found","Not Found",}}

**Review by mistaluai — CHANGES_REQUESTED** [review]
I want you to handle the following:

**Comment by BlindFl0W** [comment]
I think this should be handled in the front end. But we can do it in the DTO for sure.

**Pull request closed without merging**